torch/fx/proxy.py

```diff
@@ -112,17 +112,47 @@ def __init__(self, graph: Graph):
 class TraceError(ValueError):
     pass
 
-# Proxy objects are stand-in values for normal values in a PyTorch computation.
-# Instead of performing compute they record computation into Graph.
-# Each proxy wraps the Node instance that represents the expression that define the
-# value.
 
 class Proxy:
+    """
+    ``Proxy`` objects are ``Node`` wrappers used by the ``Tracer`` to
+    record operations seen during symbolic tracing.
+
+    The mechanism through which ``Proxy`` objects record computation is
+    ``__torch_function__``. If any custom Python type defines a method
```

Comment:
Probably a good idea to also link the documentation for `__torch_function__`: https://pytorch.org/docs/stable/notes/extending.html#extending-torch